builtin/game/mod_profiling.lua

```diff
@@ -85,272 +83,177 @@ function mod_statistics.update_statistics(dtime)
-			math.min(modtime, mod_statistics.stats[modname].min_us)
-		mod_statistics.stats[modname].max_us =
-			math.max(modtime, mod_statistics.stats[modname].max_us)
-		--TODO find better algorithm
-		mod_statistics.stats[modname].avg_us =
-			mod_statistics.stats[modname].avg_us * 0.99 + modtime * 0.01
-			
-		--update percentage values
-		local cur_per = (modtime/mod_statistics.step_total) * 100
-		mod_statistics.stats[modname].min_per =
-			math.min(mod_statistics.stats[modname].min_per, cur_per)
-			
-		mod_statistics.stats[modname].max_per =
-			math.max(mod_statistics.stats[modname].max_per, cur_per)
-			
-		--TODO find better algorithm
-		mod_statistics.stats[modname].avg_per =
-			mod_statistics.stats[modname].avg_per * 0.99 + cur_per * 0.01
+
+		-- Per mod statistics
+		-- Update microsecond values
+		stats.min_us = math.min(mod_time, stats.min_us)
+		stats.max_us = math.max(mod_time, stats.max_us)
+		-- TODO: Find better algorithm
+		stats.avg_us = stats.avg_us * 0.99 + mod_time * 0.01
+
+		-- Update percentage values
+		local cur_per = (mod_time/mod_statistics.step_total) * 100
+		stats.min_per = math.min(stats.min_per, cur_per)
+
+		stats.max_per = math.max(stats.max_per, cur_per)
+
+		-- TODO: Find better algorithm
+		stats.avg_per = stats.avg_per * 0.99 + cur_per * 0.01
 	end
-	
-	--update "total"
-	mod_statistics.stats["total"].min_us =
-		math.min(mod_statistics.step_total, mod_statistics.stats["total"].min_us)
-	mod_statistics.stats["total"].max_us =
-		math.max(mod_statistics.step_total, mod_statistics.stats["total"].max_us)
-	--TODO find better algorithm
-	mod_statistics.stats["total"].avg_us =
-		mod_statistics.stats["total"].avg_us * 0.99 +
+
+	-- Update total
+	stats_t.min_us = math.min(mod_statistics.step_total, stats_t.min_us)
+	stats_t.max_us = math.max(mod_statistics.step_total, stats_t.max_us)
+	-- TODO: Find better algorithm
+	stats_t.avg_us = stats_t.avg_us * 0.99 +
 		mod_statistics.step_total * 0.01
-	
+
 	mod_statistics.step_total = 0
 end
 
---------------------------------------------------------------------------------
-local function build_callback(log_id, fct)
-	return function( toregister )
-		local modname = core.get_current_modname()
-		
-		fct(function(...)
-			local starttime = core.get_us_time()
-			-- note maximum 10 return values are supported unless someone finds
-			-- a way to store a variable lenght return value list
-			local r0, r1, r2, r3, r4, r5, r6, r7, r8, r9 = toregister(...)
-			local delta = core.get_us_time() - starttime
-			mod_statistics.log_time(log_id, modname, delta)
-			return r0, r1, r2, r3, r4, r5, r6, r7, r8, r9
-			end
-		)
+
+local function elipsize(str, len)
+	if #str > len then
+		return "…"..str:sub(-len + 1)
 	end
+	return str
 end
 
---------------------------------------------------------------------------------
-function profiling_print_log(cmd, filter)
 
-	print("Filter:" .. dump(filter))
+function profiling_print_log(player_name, filter)
+	local function log_line(...)
```

Comment:
It doesn't take a format string, the fact that it uses string.format is a implementation detail of the function.